src/util/text-utils.cc

```diff
@@ -243,12 +264,38 @@ bool ConvertStringToReal(const std::string &str,
     return true;
 #endif  // _MSC_VER
 
-  float f = KALDI_STRTOF(this_str, &end);
-  if (end != this_str)
-    while (isspace(*end)) end++;
-  if (end == this_str || *end != '\0' || errno != 0)
-    return false;
-  *out = f;
+  std::istringstream i(str);
+
+  if (!(i >> *out)) {
+    // Number conversion failed.
+    // I know it's not a numeric value.
+    // These lines are here just do check if str is nan or inf
+    const char *this_str = str.c_str();
+    char *end = NULL;
+    errno = 0;
+
+    float d = KALDI_STRTOF(this_str, &end);
```

Comment:
It seems to me that if we're going to change the code in this way, it would be better to get rid of the strtof and strtod functions entirely, and remove the ifdefs related to them, since they are a compatibility headache, plus it's not clear to me whether their behavior would depend on local in these cases.

Could you try writing code to handle those cases without using strtof/strtod?
Based on
http://stackoverflow.com/questions/11420263/is-it-possible-to-read-infinity-or-nan-values-using-input-streams
I think you should make sure to test for inf, -inf, +inf, nan, -nan, +nan, and also case variants with NaN and Inf.  The testing may need to be extended to cover more cases than before.